### ISSUE TITLE
Use with Proxy

### ISSUE THREAD

**Issue by baileytincher** [alexa/alexa-skills-kit-sdk-for-nodejs#560]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

Many organizations can only use CLI tools like this through their internal proxy. Currently there is no documentation on how to do this, while this feature is available to the AWS CLI.

## Expected Behavior

The tool should be able to use an HTTP proxy as does the AWS CLI https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-proxy.html

## Current Behavior

```sh
ask init
...
[Error]: Cannot obtain access token. Error: self signed certificate in certificate chain
```

**Comment by ShenChen93** [comment]
Hi @baileytincher 

Sorry for responding late and thanks for the feature request. We tried to implement this feature in current version of CLI, but this require us to replace the simple-oauth2 module as this module doesn't provide developer ability to use proxy, reference: https://github.com/lelylan/simple-oauth2/issues/16. After discussion within our team, I am sorry to info you that we  are not going to support this feature recently. However, this feature request is a good call, and we have put it on the feature plan. I will keep this issue open to track this feature. Please keep tuned :)

Regards,
Shen

**Comment by baileytincher** [comment]
Hi @ShenChen-Amazon thank you for looking into this. 

Do you have any recommendations on workarounds? The company I am working for is a large enterprise with tight security measures in place. We cannot currently use the ASK CLI due to these proxy related issues, and without it have no means of properly developing for Alexa. I have to imagine that any similar company in a regulated industry would have the same dilemma.

**Comment by ShenChen93** [comment]
Hi @baileytincher 

Thanks again for reaching out. I see what you mean and it's defenitely a valid point. We have actively started looking into the code. Thanks for your patient and I will update the thread once we have progress.

**Comment by ShenChen93** [comment]
Hi, @baileytincher 

We released the CLI V1.7.10 at July 31st, and add new feature to support proxy usage. User now can set their proxy by setting the Environment variable. Let me know if you have any problems :)

**Comment by baileytincher** [comment]
> Hi, @baileytincher
> 
> We released the CLI V1.7.10 at July 31st, and add new feature to support proxy usage. User now can set their proxy by setting the Environment variable. Let me know if you have any problems :)

This is fantastic, thank you!

Ironically the last day of my internship working on this ended this morning, so I cannot personally verify. I am going to forward this back to my team though and hopefully they can test it out and get back to you!

Appreciate the quick response though, this is exactly what they needed

**Comment by ShenChen93** [comment]
Hi, @baileytincher 

I am sorry this feature block your intern project. Thank you again for posting this request. I will close this issue and please feel free to reopen if you got any problems.

Thanks,
Shen
